File: src/main/java/me/oriharel/machinery/data/ChancableList.java
package me.oriharel.machinery.data;

import me.oriharel.machinery.machine.MachineResourceGetProcess;

import java.util.ArrayList;

public class ChancableList<T extends ChanceableOperation<?, MachineResourceGetProcess>> extends ArrayList<T> implements ChanceableOperation<T,
        MachineResourceGetProcess> {
    @Override
    public void executeChanceOperation(MachineResourceGetProcess param1) {
        for (T t : this) {
            t.executeChanceOperation(param1);
        }
    }

    @Override
    public T getChanced() {
        return null;
    }
}

File: src/main/kotlin/me/oriharel/machinery/data/ChancableList.kt
package me.oriharel.machinery.data

import me.oriharel.machinery.machine.MachineResourceGetProcess
import java.util.*

class ChancableList<T : ChanceableOperation<*, MachineResourceGetProcess?>?> : ArrayList<T>(), ChanceableOperation<T, MachineResourceGetProcess> {
    override fun getChanced(lootModifier: Double): T? {
        return null
    }

    override fun executeChanceOperation(param1: MachineResourceGetProcess?, lootModifier: Double) {
        for (t in this) {
            t!!.executeChanceOperation(param1, lootModifier)
        }
    }
}